hal/src/rtl872x/deviceid_hal.cpp

```diff
@@ -165,6 +165,21 @@ int hal_get_device_hw_model(uint32_t* model, uint32_t* variant, void* reserved)
     return SYSTEM_ERROR_NONE;
 }
 
+int hal_get_device_hw_info(hal_device_hw_info* info, void* reserved) {
+    CHECK_TRUE(info, SYSTEM_ERROR_INVALID_ARGUMENT);
+    // HW Data format: | NCP_ID (LSB) | HW_VERSION | HW Feature Flags |
+    //                 |    byte 0    |   byte 1   |    byte 2/3      |
+    uint8_t hw_data[4] = {};
```

Comment:
It would be helpful to return the NCP ID in the `hal_device_hw_info` struct here as well